grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -271,4 +271,68 @@ class CycleCountService {
         }
         return cycleCounts.collect { CycleCountDto.createFromCycleCountItems(it.cycleCountItems) }
     }
+
+    CycleCountDto submitCount(CycleCountSubmitCountCommand command) {
+        command.cycleCount.cycleCountItems.each { CycleCountItem cycleCountItem ->
+            updateCycleCountItemForSubmit(cycleCountItem, command.refreshQuantityOnHand, command.failOnOutdatedQuantity)
+            determineCycleCountItemStatus(cycleCountItem, command.requireRecountOnDiscrepancy)
+        }
+        if (command.cycleCount.status == CycleCountStatus.COUNTED) {
+            createCycleCountTransaction(command.cycleCount)
+        }
+        return CycleCountDto.createFromCycleCountItems(command.cycleCount.cycleCountItems)
+    }
+
+    void updateCycleCountItemForSubmit(CycleCountItem cycleCountItem, boolean refreshQuantityOnHand, boolean failOnOutdatedQuantity) {
+        Integer currentQuantityOnHand =
+            productAvailabilityService.getQuantityOnHandInBinLocation(cycleCountItem.inventoryItem, cycleCountItem.location)
+        if (refreshQuantityOnHand) {
+            cycleCountItem.quantityOnHand = currentQuantityOnHand
+        }
+        if (failOnOutdatedQuantity && cycleCountItem.quantityOnHand != currentQuantityOnHand) {
+            throw new IllegalArgumentException("Quantity on hand for a cycle count item is no longer up to date")
+        }
+    }
+
+    void determineCycleCountItemStatus(CycleCountItem cycleCountItem, boolean requireRecountOnDiscrepancy) {
+        if (cycleCountItem.quantityOnHand == cycleCountItem.quantityCounted) {
+            cycleCountItem.status = CycleCountItemStatus.COUNTED
+            return
+        }
+        if (!requireRecountOnDiscrepancy) {
+            cycleCountItem.status = CycleCountItemStatus.APPROVED
+            return
+        }
+        // FIXME: INVESTIGATING = DISCREPANCY_TO_RESOLVE???
+        cycleCountItem.status = CycleCountItemStatus.INVESTIGATING
```

Comment:
the same case as above - it was meant to be set as `DISCREPANCY_TO_RESOLVE`, but again - we don't have such a status, so is it equivalent to `INVESTIGATING`?

Reply:
I think we want COUNTED. We use COUNTED to mean there is a discrepancy but recounting hasn't been started yet (once the recount starts, the status will be INVESTIGATING).

I'm not the hugest fan of this because to me the word "counted" doesn't imply a discrepancy, but that's what we have currently